azure-configurationservice/tests/test_azure_configuration_client.py

```diff
@@ -0,0 +1,601 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import os
+import datetime
+from copy import copy
+import pytest
+
+
+from msrestazure.azure_exceptions import CloudError
+from azure.configuration import AzureConfigurationClient
+from azure.configuration import ConfigurationSetting
+from devtools_testutils import AzureMgmtTestCase
+
+import conftest
+
+
+class AzConfigTestData:
+    def __init__(self):
+        self.key_uuid = None
+        self.label1 = None
+        self.label2 = None  # contains reserved chars *,\
+        self.label_uuid = None
+        self.label1_data = []
+        self.label2_data = []
+        self.no_label_data = []
+
+
+class AzConfigurationClientTest(AzureMgmtTestCase):
+    def setUp(self):
+        self.working_folder = os.path.dirname(__file__)
+        super(AzConfigurationClientTest, self).setUp()
+
+        connection_str = os.environ["AZ_CONFIG_CONNECTION"]
+        self.app_config_client = AzureConfigurationClient(connection_str)
+        self.test_data = conftest.setup_data()
+
+    def tearDown(self):
+        conftest.teardown_data(self.test_data)
+
+    # method: add_configuration_setting
+    def test_add_configuration_setting(self):
+
+        kv = ConfigurationSetting()
+        kv.key = "unit_test_key_add_" + self.test_data.key_uuid
+        kv.label = self.test_data.label1
+        kv.value = "test value"
+        kv.content_type = "test content type"
+        kv.tags = {"tag1": "tag1", "tag2": "tag2"}
+        kv.etag = "not needed"  # this etag will not be processed.
+        created_kv = self.app_config_client.add_configuration_setting(kv)
+        self.app_config_client.delete_configuration_setting(kv.key, kv.label)
+        assert (
+            created_kv.label == kv.label
+            and kv.value == kv.value
+            and created_kv.content_type == kv.content_type
+            and created_kv.tags == kv.tags
+        )
+        assert (
+            created_kv.etag is not None
+            and created_kv.last_modified is not None
+            and created_kv.locked is False
+        )
+
+    def test_add_existing_configuration_setting(self):
+        kv = ConfigurationSetting()
+        compare_kv = self.test_data.label1_data[0]
+        kv.key = compare_kv.key
+        kv.label = compare_kv.label
+        kv.value = compare_kv.value
+        kv.content_type = compare_kv.content_type
+
+        with pytest.raises(CloudError):
+            self.app_config_client.add_configuration_setting(kv)
+
+    def test_add_configuration_setting_no_label(self):
```

Comment:
what is the purpose of this test? 
- make sure the URL get successfully created if there is no label? 
- make sure the service supports the creation of a setting without a label

Reply:
To make sure we passed the correct data and get correct response. Main reason is previously in python the add method had the kv object, key and label (default None or '%00' at different times) as inputs. So I developed test methods to have label and have no label as input. Later key and label were removed from the API. Since this no-lable test doesn't hurt anything, I keep them.

Reply:
I see. Right now they are not adding value. Just think to maintain them it involves more work and also the time it takes running all the test is going to be incremented by tests that we don't need. 
I will prefer to have